### PR TITLE
[ci] support artifact mount for all containers

### PR DESCRIPTION
**Pull request by can-anyscale**
Support artifact mount for all containers (linux + windows)

Test:
- CI

### CODE REVIEW

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -25,8 +25,15 @@ if [ -d "/tmp/artifacts" ]; then
     find /tmp/artifacts -print || true
 fi
 
-echo "Creating var/ artifact directory..."
-docker run --rm -v /tmp/artifacts:/artifact-mount alpine:latest /bin/sh -c 'chown -R 2000 /artifact-mount/' || true
+echo "Creating artifact directory..."
```
Comment (aslonnie) [inline .buildkite/hooks/pre-command:28]: move the echo into the if too?

File: ci/ray_ci/windows_container.py [hunk 2]
```diff
@@ -37,3 +37,9 @@ def get_run_command_extra_args(
     ) -> List[str]:
         assert not gpu_ids, "Windows does not support gpu ids"
         return []
+
+    def get_artifact_mount_host(self) -> str:
+        return "c:\\tmp\\artifacts"
```
Comment (aslonnie) [inline ci/ray_ci/windows_container.py:42]: nit: `C:\\` (capital `C`) maybe?

File: ci/ray_ci/container.py [hunk 3]
```diff
@@ -99,3 +106,11 @@ def get_run_command_extra_args(
...
     ) -> List[str]:
         pass
+
+    @abc.abstractmethod
+    def get_artifact_mount_host(self) -> str:
+        pass
+
+    @abc.abstractmethod
+    def get_artifact_mount_container(self) -> str:
+        pass
```
Comment (aslonnie) [inline ci/ray_ci/container.py:116]: make it one method that returns a tuple?

**Review by aslonnie — APPROVED** [review]

File: .buildkite/hooks/pre-command (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/hooks/pre-command:28]: the echo applies for both if clauses (linux vs. windows)

File: ci/ray_ci/windows_container.py (hunk 2 above)
Comment (can-anyscale) [reply ci/ray_ci/windows_container.py:42]: totally

File: ci/ray_ci/container.py (hunk 3 above)
Comment (can-anyscale) [reply ci/ray_ci/container.py:116]: there are some use cases later on that I need one and not the other

Comment (aslonnie) [reply ci/ray_ci/container.py:116]: in those cases you can just drop the other return value?

the two values are so

Comment (can-anyscale) [reply ci/ray_ci/container.py:116]: sound good, let's do that

**Pull request merged**